File: app/src/main/java/com/bkjcb/rqapplication/stationCheck/fragment/CheckItemResultFragment.java
package com.bkjcb.rqapplication.stationCheck.fragment;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.RadioGroup;

import com.bkjcb.rqapplication.R;
import com.bkjcb.rqapplication.base.fragment.BaseLazyFragment;
import com.bkjcb.rqapplication.stationCheck.model.CheckItem;

/**
 * Created by DengShuai on 2020/1/7.
 * Description :
 */
public class CheckItemResultFragment extends BaseLazyFragment implements RadioGroup.OnCheckedChangeListener {

    private View view;
    private CheckItem checkItem;
    private EditText mItemRecord;
    private RadioButton ok;
    private RadioButton failure;

    public void setCheckItem(CheckItem checkItem) {
        this.checkItem = checkItem;
    }

    public static CheckItemResultFragment newInstance(CheckItem item) {
        CheckItemResultFragment fragment = new CheckItemResultFragment();
        fragment.setCheckItem(item);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        if (view == null) {
            view = inflater.inflate(R.layout.fragment_check_item_result, null);
            mItemRecord = view.findViewById(R.id.item_record);
            mItemRecord.setText(checkItem.beizhu);
            RadioGroup radioGroup = view.findViewById(R.id.check_result_radioGroup);
            RadioGroup needRadioGroup = view.findViewById(R.id.check_result_radioGroup_need);
            ok = view.findViewById(R.id.check_result_radio_ok);
            failure = view.findViewById(R.id.check_result_radio_failure);
           RadioButton need = view.findViewById(R.id.check_result_radio_need);
            RadioButton notNeed = view.findViewById(R.id.check_result_radio_donotneed);
            if (!TextUtils.isEmpty(checkItem.jianchajieguo)) {
                if ("合格".equals(checkItem.jianchajieguo)) {
                    radioGroup.check(R.id.check_result_radio_ok);
                    changeTextColor(true);
                } else {
                    radioGroup.check(R.id.check_result_radio_failure);
                    changeTextColor(false);
                }
            }
            if (!TextUtils.isEmpty(checkItem.tijiaobaogao)) {
                if ("需要整改报告".equals(checkItem.tijiaobaogao)) {
                    needRadioGroup.check(R.id.check_result_radio_need);
                } else {
                    needRadioGroup.check(R.id.check_result_radio_donotneed);
                }
            }
            if (checkItem.status <3) {
                radioGroup.setOnCheckedChangeListener(this);
                needRadioGroup.setOnCheckedChangeListener(this);
            }else {
                mItemRecord.setEnabled(false);
                ok.setEnabled(false);
                failure.setEnabled(false);
                need.setEnabled(false);
                notNeed.setEnabled(false);
            }
        }
        return view;
    }

    @Override
    public void onStop() {
        super.onStop();
    }

    @Override
    public void onCheckedChanged(RadioGroup group, int checkedId) {
        if (group.getId() == R.id.check_result_radioGroup) {
            if (checkedId == R.id.check_result_radio_ok) {
                checkItem.jianchajieguo = "合格";
                changeTextColor(true);
            } else {
                checkItem.jianchajieguo = "不合格";
                changeTextColor(false);
            }
        } else {
            if (checkedId == R.id.check_result_radio_need) {
                checkItem.tijiaobaogao = "需要整改报告";
            } else {
                checkItem.tijiaobaogao = "不需要整改报告";
            }
        }
    }

    private void changeTextColor(boolean isOk) {
        if (isOk) {
            ok.setTextColor(getResources().getColor(R.color.colorText));
            failure.setTextColor(getResources().getColor(R.color.colorSecondDrayText));
        } else {
            ok.setTextColor(getResources().getColor(R.color.colorSecondDrayText));
            failure.setTextColor(getResources().getColor(R.color.colorText));
        }
    }

    public String getRemark() {
        return mItemRecord.getText().toString();
    }
}

File: app/src/main/java/com/bkjcb/rqapplication/check/fragment/CheckItemResultFragment.kt
package com.bkjcb.rqapplication.check.fragment

import android.os.Bundle
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import android.widget.RadioButton
import android.widget.RadioGroup
import com.bkjcb.rqapplication.R
import com.bkjcb.rqapplication.check.model.CheckItem
import com.bkjcb.rqapplication.fragment.BaseLazyFragment
import kotlinx.android.synthetic.main.fragment_check_item_result.*

/**
 * Created by DengShuai on 2020/1/7.
 * Description :
 */
class CheckItemResultFragment : BaseLazyFragment(), RadioGroup.OnCheckedChangeListener {
    private var checkItem: CheckItem? = null
    private var mItemRecord: EditText? = null
    private var ok: RadioButton? = null
    private var failure: RadioButton? = null

    fun setCheckItem(checkItem: CheckItem?) {
        this.checkItem = checkItem
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        if (rootView == null) {
            rootView = inflater.inflate(R.layout.fragment_check_item_result, null)
            item_record.setText(checkItem!!.beizhu)
            if (!TextUtils.isEmpty(checkItem!!.jianchajieguo)) {
                if ("合格" == checkItem!!.jianchajieguo) {
                    check_result_radioGroup.check(R.id.check_result_radio_ok)
                    changeTextColor(true)
                } else {
                    check_result_radioGroup.check(R.id.check_result_radio_failure)
                    changeTextColor(false)
                }
            }
            check_result_radioGroup.setOnCheckedChangeListener(this)
        }
        return rootView
    }


    override fun onCheckedChanged(group: RadioGroup, checkedId: Int) {
        if (checkedId == R.id.check_result_radio_ok) {
            checkItem!!.jianchajieguo = "合格"
            changeTextColor(true)
        } else {
            checkItem!!.jianchajieguo = "不合格"
            changeTextColor(false)
        }
    }

    private fun changeTextColor(isOk: Boolean) {
        if (isOk) {
            check_result_radio_ok.setTextColor(resources.getColor(R.color.colorText))
            check_result_radio_failure.setTextColor(resources.getColor(R.color.colorSecondDrayText))
        } else {
            check_result_radio_ok.setTextColor(resources.getColor(R.color.colorSecondDrayText))
            check_result_radio_failure.setTextColor(resources.getColor(R.color.colorText))
        }
    }

    val remark: String
        get() =item_record.text.toString()

    companion object {
        fun newInstances(item: CheckItem?): CheckItemResultFragment {
            val fragment = CheckItemResultFragment()
            fragment.setCheckItem(item)
            return fragment
        }
    }
}